pkg/block/local/adapter.go

```diff
@@ -367,12 +366,34 @@ func (l *Adapter) GetProperties(_ context.Context, obj block.ObjectPointer) (blo
 		}
 		return block.Properties{}, err
 	}
-	// No properties, just return that it exists
+
+	etag, err := calcFileETag(p)
+	if err != nil {
+		return block.Properties{}, err
+	}
+
 	return block.Properties{
 		LastModified: stat.ModTime(),
+		ETag:         etag,
 	}, nil
 }
 
+func calcFileETag(fileKey string) (string, error) {
+	f, err := os.Open(fileKey)
+	if err != nil {
+		return "", err
+	}
+	defer func() { _ = f.Close() }()
+	// MD5 required for ETag computation.
```

Comment:
not sure about this comment - we can use any algorithm as long as we use the same one. true we can't change it now as we already used md5 as the algorithm so far.

Reply:
It's a copy-paste of the original code.
I realllly don't mind remove it if you think it's better.